arlo_server/superadmin.py

```diff
@@ -46,3 +50,238 @@ def superadmin_jurisdictionadmin_login():
+            total_ballots_cast=old_contest.total_ballots_cast,
+            num_winners=old_contest.num_winners,
+            votes_allowed=old_contest.votes_allowed,
+        )
+
+        # this map keeps the old object around because it's needed
+        contest_map[old_contest.id] = (old_contest, new_contest)
+
+        db.session.add(new_contest)
+
+        for old_choice in old_contest.choices:
+            new_choice = ContestChoice(
+                id=new_id(),
+                contest_id=new_contest.id,
+                name=old_choice.name,
+                num_votes=old_choice.num_votes,
+            )
+
+            choice_map[old_choice.id] = new_choice
+
+            db.session.add(new_choice)
+
+        for old_roundcontest in RoundContest.query.filter_by(contest_id=old_contest.id):
+            new_roundcontest = RoundContest(
+                round_id=round_map[old_roundcontest.round_id].id,
+                contest_id=new_contest.id,
+                sample_size_options=old_roundcontest.sample_size_options,
+                end_p_value=old_roundcontest.end_p_value,
+                is_complete=old_roundcontest.is_complete,
+                sample_size=old_roundcontest.sample_size,
+            )
+
+            db.session.add(new_roundcontest)
+
+            for old_result in old_roundcontest.results:
+                new_result = RoundContestResult(
+                    round_id=round_map[old_result.round_id].id,
+                    contest_id=new_contest.id,
+                    contest_choice_id=choice_map[old_result.contest_choice_id].id,
+                    result=old_result.result,
+                )
+                db.session.add(new_result)
+
+    jurisdiction_map = {}
+
+    for old_jurisdiction in old_election.jurisdictions:
+        new_jurisdiction = Jurisdiction(
+            id=new_id(),
+            election_id=new_election.id,
+            name=old_jurisdiction.name,
+            manifest_num_ballots=old_jurisdiction.manifest_num_ballots,
+            manifest_num_batches=old_jurisdiction.manifest_num_batches,
+        )
+
+        jurisdiction_map[old_jurisdiction.id] = new_jurisdiction
+
+        if old_jurisdiction.manifest_file_id:
+            old_file = old_jurisdiction.manifest_file
+            new_file = File(
+                id=new_id(),
+                name=old_file.name,
+                contents=old_file.contents,
+                uploaded_at=old_file.uploaded_at,
+                processing_started_at=old_file.processing_started_at,
+                processing_completed_at=old_file.processing_completed_at,
+                processing_error=old_file.processing_error,
+            )
+            db.session.add(new_file)
+
+            new_jurisdiction.manifest_file_id = new_file.id
+
+        db.session.add(new_jurisdiction)
+
+        for old_ja in old_jurisdiction.jurisdiction_administrations:
+            new_user = clone_user_with_prefix(old_ja.user, "superadmin--")
```

Comment:
how will they be able to log in with this made up email address?

Reply:
yeah this was for testing purposes, and I think it's worth a more thorough review of why we want this before we merge. In particular, I think prefixing email addresses should be an option, not a forced thing.